Dockerfile

```diff
@@ -51,6 +51,48 @@ COPY LICENSE /usr/share/doc/$PRODUCT_NAME/LICENSE.txt
 COPY dist/$TARGETOS/$TARGETARCH/vault-csi-provider /bin/
 ENTRYPOINT [ "/bin/vault-csi-provider" ]
 
+# ubi build image
+# -----------------------------------
+FROM registry.access.redhat.com/ubi10/ubi-minimal:10.0-1754585875 AS build-ubi
+RUN microdnf --refresh --assumeyes upgrade ca-certificates
+
+# ubi release image
+# -----------------------------------
+FROM registry.access.redhat.com/ubi10/ubi-micro:10.0-1754556444 AS release-ubi
+
+ENV BIN_NAME=vault-csi-provider
+ARG PRODUCT_VERSION
+ARG PRODUCT_REVISION
+ARG PRODUCT_NAME=$BIN_NAME
+# TARGETARCH and TARGETOS are set automatically when --platform is provided.
+ARG TARGETOS TARGETARCH
+
+LABEL name="Vault CSI Provider" \
```

Comment:
This isn't a blocker, as I don't think it's the end of the world if this label says "Vault CSI Provider", but I think we decided that in order to more easily differentiate from our own CSI driver, we'd start calling this offering "Secrets Store CSI Provider for Vault" in more official human-readable contexts (i.e. we don't need to change the binary name/repo name). 

My PR for the VSO CSI driver docs replaces "Vault CSI Provider" with that name across all docs pages that mention it. It just makes it easier to understand that this is our contribution to someone else's CSI driver, while we also have our own.

May want to double-check this with Steve though, if you have any hesitation about us changing the official name!

Reply:
The summary/description for the image just below may also cover this confusion 🤷‍♀️

Reply:
Ok, these metadata labels are just copied from the existing alpine image above. If we're adjusting the name I think we'll also want to update the descriptions in the docker hub and Red Hat registries.

Side note, "Secrets Store CSI Provider for Vault" sounds a little backwards to me? Should it be something more like Vault Secrets Store CSI Provider? i.e. Vault is the thing being provided, not provided for in my mind.